trieste/acquisition/function/entropy.py

```diff
@@ -617,3 +621,231 @@ def __call__(self, x: TensorType) -> TensorType:
+        num_samples: int = 5,
+        grid_size: int = 1000,
+        min_value_sampler: None = None,
+    ):
+        ...
+
+    @overload
+    def __init__(
+        self: "MUMBO[MUMBOModelType]",
+        search_space: SearchSpace,
+        num_samples: int = 5,
+        grid_size: int = 1000,
+        min_value_sampler: Optional[ThompsonSampler[MUMBOModelType]] = None,
+    ):
+        ...
+
+    def __init__(
+        self,
+        search_space: SearchSpace,
+        num_samples: int = 5,
+        grid_size: int = 1000,
+        min_value_sampler: Optional[ThompsonSampler[MUMBOModelType]] = None,
+    ):
+        super().__init__(search_space, num_samples, grid_size, min_value_sampler)
+
+    def prepare_acquisition_function(
+        self,
+        model: MUMBOModelType,
+        dataset: Optional[Dataset] = None,
+    ) -> AcquisitionFunction:
+        """
+        :param model: The multifidelity model.
+        :param dataset: The data from the observer.
+        :return: The max-value entropy search acquisition function modified for objective
+            minimisation. This function will raise :exc:`ValueError` or
+            :exc:`~tf.errors.InvalidArgumentError` if used with a batch size greater than one.
+        :raise tf.errors.InvalidArgumentError: If ``dataset`` is empty.
+        """
+        tf.debugging.Assert(dataset is not None, [])
+        dataset = cast(Dataset, dataset)
+        tf.debugging.assert_positive(len(dataset), message="Dataset must be populated.")
+        min_value_samples = self.get_min_value_samples_on_top_fidelity(model, dataset)
+        return mumbo(model, min_value_samples)
+
+    def update_acquisition_function(
+        self,
+        function: AcquisitionFunction,
+        model: MUMBOModelType,
+        dataset: Optional[Dataset] = None,
+    ) -> AcquisitionFunction:
+        """
+        :param function: The acquisition function to update.
+        :param model: The model.
+        :param dataset: The data from the observer.
+        """
+        tf.debugging.Assert(dataset is not None, [])
+        dataset = cast(Dataset, dataset)
+        tf.debugging.assert_positive(len(dataset), message="Dataset must be populated.")
+        min_value_samples = self.get_min_value_samples_on_top_fidelity(model, dataset)
+        function.update(min_value_samples)  # type: ignore
+        return function
+
+    def get_min_value_samples_on_top_fidelity(
+        self, model: MUMBOModelType, dataset: Dataset
+    ) -> TensorType:
+        """
+        :param model: The model.
+        :param dataset: The data from the observer.
+        """
+        query_points = self._search_space.sample(num_samples=self._grid_size)
+        tf.debugging.assert_same_float_dtype([dataset.query_points, query_points])
+        query_points = tf.concat([dataset.query_points, query_points], 0)
+        query_points_on_top_fidelity = tf.concat(
+            [query_points[:, :-1], tf.ones_like(query_points[:, -1:])], -1
+        )
```

Comment:
Couple of things here, I think we should use the function designed for this, and I think we have assumed 2 fidelities, 1 is not guaranteed to be the top fidelity.

```suggestion
        query_points_on_top_fidelity = add_fidelity_column(query_points[:, :-1], model.num_fidelities -1)
```

Reply:
Good catch!

Reply:
So interestingly changing this to use `fidelity=model.num_fidelities-1` rather than `fidelity=1` causes `test_mumbo_builder_builds_min_value_samples` to fail on `assert max(min_value_samples) < min(fmean)`. Any idea what's going on?

Reply:
I'm not sure, sorry